### ISSUE TITLE
Introduce `mip` level fixes

### ISSUE THREAD

**Issue by schlunma** [ESMValGroup/ESMValCore#1094]
As proposed by @tomaslovato in #1069, it would be nice to have fixes at `mip` level for each dataset. For example, this fix

```py
class Amon(Fix):
    """Fixes for all `Amon` variables."""
```

would be applied to all `Amon` variables of the given dataset. This would probably avoid lots of code duplication.

The implementation should be fairly easy as pointed out by @tomaslovato [here](https://github.com/ESMValGroup/ESMValCore/issues/1069#issuecomment-822432524).

I really like this idea! What say you @valeriupredoi @zklaus @jvegasbsc? We could discuss this in the general discussion of the workshop if necessary.

**Comment by jvegasbsc** [comment]
It will be fairly easy to implement, as we are already passing the mip to the get_fixes function. Just go from

```python
for fix_name in (short_name, 'allvars'):
    try:
        fixes.append(classes[fix_name](vardef))
    except KeyError:
        pass
```

```python
for fix_name in (short_name, mip.lower(), 'allvars'):
    try:
        fixes.append(classes[fix_name](vardef))
    except KeyError:
        pass
```

**Comment by tomaslovato** [comment]
@jvegasbsc I definitely agree  🎉  (0837102dba5)

**Comment by schlunma** [comment]
If nobody has objections, I will open a PR based on `ce2coast` branch :+1:

**Comment by tomaslovato** [comment]
@schlunma note that in the same branch there is also a tiny fix 3dad5db proposed for #1018 that I already discussed with @valeriupredoi

**Comment by schlunma** [comment]
Yes, I already saw that. For now I cherry-picked only the parts of your branch that are necessary for the `mip`-level fixes (and corresponding tests). I think it's best to include the other changes discussed in #1069 in a separate pull request.
